legacy/src/app/controllers/node_details.js

```diff
@@ -6,6 +6,38 @@
 
 import { HardwareType, NodeTypes } from "../enum";
 
+// Convert MiB to GiB value if over 1024 and round to 4 significant figures.
+const formatNumaMemory = (mib) => {
+  if (mib >= 1024) {
+    return `${Number((mib / 1024).toPrecision(4)).toString()} GiB`;
+  }
+  return `${Number(mib.toPrecision(4)).toString()} MiB`;
+};
+
+// Convert array of numbers into range strings,
+// e.g [0, 1, 2, 4, 6, 7] => ["0-2", "4", "6-7"]
+const getRanges = (array) => {
+  const ranges = [];
+  let rangeStart;
+  let rangeEnd;
+
+  for (let i = 0; i < array.length; i++) {
+    rangeStart = array[i];
+    rangeEnd = rangeStart;
+    // Keep incrementing rangeEnd while its a consecutive number.
+    while (parseInt(array[i + 1]) - parseInt(array[i]) === 1) {
+      rangeEnd = parseInt(array[i + 1]);
+      i++;
+    }
+    ranges.push(
+      rangeStart === rangeEnd ? `${rangeStart}` : `${rangeStart}-${rangeEnd}`
```

Comment:
```... ? `${rangeStart}` : ...``` could be `... ? rangeStart : ...` or is that to convert it to a string?

Reply:
Yeah I've forced them all to be strings so you don't end up with mixed types in the array, like `[0, "1-2", 3, "4-10"]` etc.